dspy/retrieve/retrieve.py

```diff
@@ -16,13 +18,20 @@ def single_query_passage(passages):
         passages_dict["passages"] = passages_dict.pop("long_text")
     return Prediction(**passages_dict)
 
+@lru_cache(maxsize=None)
+def warn_once(msg: str):
+    logging.warning(msg)
```

Comment:
We have standardized the logging pattern to be: 

```
import logging

logger = logging.getLogger(__name__)
logger.info(...)
```

For example: https://github.com/stanfordnlp/dspy/blob/943ee02d4cf9a630c89136ff6dc6b9f6ebf805dd/dspy/utils/callback.py#L12